### PR TITLE
[TASK] Switch to confvals for rte

### PR DESCRIPTION
**Pull request by linawolf**
* use confvals
* correct indentation

releases: main, 12.4, 11.5

### CODE REVIEW

File: Documentation/PageTsconfig/Rte.rst [hunk 1]
```diff
@@ -42,37 +42,44 @@ will usually be used. This page covers only the general configuration, for
 more information about configuring EXT:rte_ckeditor, see the
 :ref:`rte_ckeditor configuration <ext_rte_ckeditor:configuration>`.
 
+.. contents::
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:45]:
```suggestion
..  contents::
```

File: Documentation/PageTsconfig/Rte.rst [hunk 2]
```diff
@@ -95,262 +102,179 @@ Summary:
...
-
-   Description
-         If set, the selection of link target is removed from the link
-         insertion/update dialog.
-
-         Default : 0
+..  confval:: buttons.link.targetSelector.disabled
+    :name: rte-buttons-link-targetselector-disabled
+    :type: boolean
+    :Default: false
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:157]:
```suggestion
    :Default: 0
```

This is page TSconfig, not PHP. I assume, "false" may be interpreted as true as it is a non-empty string.

File: Documentation/PageTsconfig/Rte.rst [hunk 3]
```diff
@@ -95,262 +102,179 @@ Summary:
...
-   Description
-         If set, the specification of a page id, without using the page tree,
-         is enabled in the link insertion/update dialog.
-
-         Note: This feature is intended for authors who have to deal with a
-         very large page tree. Note that the feature is disabled by default.
+..  confval:: buttons.link.pageIdSelector.enabled
+    :name: rte-buttons-link-pageidselector-enabled
+    :type: boolean
+    :Default: false
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:170]:
```suggestion
    :Default: 0
```

File: Documentation/PageTsconfig/Rte.rst [hunk 4]
```diff
@@ -95,262 +102,179 @@ Summary:
...
-         boolean
-
-   Description
-         If set, an additional field is enabbled in the link insertion/update
-         dialogue allowing authors to specify query parameters to be added on
-         the link
+..  confval:: buttons.link.queryParametersSelector.enabled
+    :name: rte-buttons-link-queryparametersselector-enabled
+    :type: boolean
+    :Default: false
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:186]:
```suggestion
    :Default: 0
```

File: Documentation/PageTsconfig/Rte.rst [hunk 5]
```diff
@@ -95,262 +102,179 @@ Summary:
...
-
-   Property
-         buttons.link.relAttribute.enabled
-
-   Data type
-         boolean
+..  confval:: buttons.link.relAttribute.enabled
+    :name: rte-buttons-link-relattribute-enabled:
+    :type: boolean
+    :Default: false
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:200]:
```suggestion
    :Default: 0
```

File: Documentation/PageTsconfig/Rte.rst [hunk 6]
```diff
@@ -391,16 +316,17 @@ Notice how many properties relate to specific transformations only! Also notice
...
 
-.. index::
-   RTE; Classes allowed
+..  index::
+    RTE; Classes allowed
+..  _rte-proc-allowedClasses:
 
 allowedClasses
-~~~~~~~~~~~~~~
+~~~~~~~~~~~~~
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:324]:
```suggestion
~~~~~~~~~~~~~~
```

File: Documentation/PageTsconfig/Rte.rst [hunk 7]
```diff
@@ -567,18 +498,20 @@ HTMLparser_db
 
     For the detailed list of properties, see the :ref:`section of the TypoScript reference <t3tsref:htmlparser>`.
 
-    .. note::
+    ..  note::
 
        This configuration is similar in frontend TypoScript and Page TSconfig.
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:503]: Indentation missing.

**Review by brotkrueml — CHANGES_REQUESTED** [review]

File: Documentation/PageTsconfig/Rte.rst [hunk 8]
```diff
@@ -567,18 +498,20 @@ HTMLparser_db
...
        This is why single properties can be looked up in the TypoScript reference.
 
        Also note the :ref:`HTMLparser <t3tsref:htmlparser>` options :code:`keepNonMatchedTags`
        and :code:`htmlSpecialChars` are *not* observed. They are preset internally.
```
Comment (linawolf) [inline Documentation/PageTsconfig/Rte.rst:507]:
```suggestion
        This configuration is similar in frontend TypoScript and Page TSconfig.
        This is why single properties can be looked up in the TypoScript reference.

        Also note the :ref:`HTMLparser <t3tsref:htmlparser>` options :code:`keepNonMatchedTags`
        and :code:`htmlSpecialChars` are *not* observed. They are preset internally.
```

File: Documentation/PageTsconfig/Rte.rst [hunk 9]
```diff
@@ -414,151 +340,156 @@ allowedClasses
...
+    RTE; Tags allowed
+..  _rte-proc-allowTags:
 
 allowTags
 ~~~~~~~~~
 
-:aspect:`Datatype`
-    string with comma separated values
+..  confval:: proc.allowTags
+    :name: rte-proc-allowTags:
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:352]:
```suggestion
    :name: rte-proc-allowTags
```

File: Documentation/PageTsconfig/Rte.rst [hunk 10]
```diff
@@ -95,262 +102,179 @@ Summary:
...
 
-.. _buttons-link-relattribute-enabled:
+..  _buttons-link-relattribute-enabled:
 
 buttons.link.relAttribute.enabled
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. container:: table-row
+..  confval:: buttons.link.relAttribute.enabled
+    :name: rte-buttons-link-relattribute-enabled:
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:198]:
```suggestion
    :name: rte-buttons-link-relattribute-enabled
```

File: Documentation/PageTsconfig/Rte.rst [hunk 11]
```diff
@@ -95,262 +102,179 @@ Summary:
...
-   Property
-         buttons.link.properties.class.allowedClasses
-
-   Data type
-         list of id-strings
-
-   Description
-         Classes available in the Insert/Modify link dialogue.
+..  confval:: buttons.link.properties.class.allowedClasses
+    :name: rte-buttons-link-properties-class-allowedclasses:
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:212]:
```suggestion
    :name: rte-buttons-link-properties-class-allowedclasses
```

File: Documentation/PageTsconfig/Rte.rst [hunk 12]
```diff
@@ -95,262 +102,179 @@ Summary:
...
 
-.. container:: table-row
-
-   Property
-         buttons.link.properties.class.required
-
-   Data type
-         boolean
+..  confval:: buttons.link.properties.class.required
+    :name: rte-buttons-link-properties-class-required:
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:223]:
```suggestion
    :name: rte-buttons-link-properties-class-required
```

File: Documentation/PageTsconfig/Rte.rst [hunk 13]
```diff
@@ -95,262 +102,179 @@ Summary:
...
 
-.. container:: table-row
-
-   Property
-         buttons.link.[ *type* ].properties.class.required
-
-   Data type
-         boolean
+..  confval:: buttons.link.[type].properties.class.required
+    :name: rte-buttons-link-type-properties-class-required:
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:235]:
```suggestion
    :name: rte-buttons-link-type-properties-class-required
```

File: Documentation/PageTsconfig/Rte.rst [hunk 14]
```diff
@@ -95,262 +102,179 @@ Summary:
...
 
 buttons.link.properties.target.default
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. container:: table-row
-
-   Property
-         buttons.link.properties.target.default
+..  confval:: buttons.link.properties.target.default
+    :name: rte-buttons-link-properties-target-default:
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:248]:
```suggestion
    :name: rte-buttons-link-properties-target-default
```

File: Documentation/PageTsconfig/Rte.rst [hunk 15]
```diff
@@ -95,262 +102,179 @@ Summary:
...
-
-   Data type
-         string
-
-   Description
-         Specifies a default target for links of the given type.
-         Possible types are: page, file, url, mail, spec. More types may be
-         provided by extensions.
+..  confval:: buttons.link.[type].properties.target.default
+    :name: rte-buttons-link-type-properties-target-default:
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:260]:
```suggestion
    :name: rte-buttons-link-type-properties-target-default
```

File: Documentation/PageTsconfig/Rte.rst [hunk 16]
```diff
@@ -95,262 +102,179 @@ Summary:
...
-contentsLanguageDirection
-~~~~~~~~~~~~~~~~~~~~~~~~~
-
-
-:aspect:`Datatype`
-    :typoscript:`rtl` or :typoscript:`ltr`
-
-:aspect:`Description`
+..  confval:: config.contentsLanguageDirection
+    :name: rte-config-contentsLanguageDirection:
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:276]:
```suggestion
    :name: rte-config-contentsLanguageDirection
```

File: Documentation/PageTsconfig/Rte.rst [hunk 17]
```diff
@@ -391,16 +316,17 @@ Notice how many properties relate to specific transformations only! Also notice
...
+    RTE; Classes allowed
+..  _rte-proc-allowedClasses:
 
 allowedClasses
 ~~~~~~~~~~~~~~
 
-:aspect:`Datatype`
-    string with comma separated values
+..  confval:: proc.allowedClasses
+    :name: rte-proc-allowedClasses:
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:327]:
```suggestion
    :name: rte-proc-allowedClasses
```

File: Documentation/PageTsconfig/Rte.rst [hunk 18]
```diff
@@ -414,151 +340,156 @@ allowedClasses
...
+    RTE; tags denyed
+..  _rte-proc-denyTags:
 
 denyTags
 ~~~~~~~~
 
-:aspect:`Datatype`
-    string with comma separated values
+..  confval:: proc.denyTags
+    :name: rte-proc-denyTags:
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:410]:
```suggestion
    :name: rte-proc-denyTags
```

File: Documentation/PageTsconfig/Rte.rst [hunk 19]
```diff
@@ -414,151 +340,156 @@ allowedClasses
...
+    RTE; HTMLparser DB entry
+..  _rte-proc-entryHTMLparser_db:
 
 entryHTMLparser_db
 ~~~~~~~~~~~~~~~~~~
 
-:aspect:`Datatype`
-    boolean / :ref:`HTMLparser <t3tsref:htmlparser>`
+..  confval:: proc.entryHTMLparser_db
+    :name: rte-proc-entryHTMLparser_db:
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:425]:
```suggestion
    :name: rte-proc-entryHTMLparser_db
```

File: Documentation/PageTsconfig/Rte.rst [hunk 20]
```diff
@@ -414,151 +340,156 @@ allowedClasses
...
+    RTE; HTMLparser RTE entry
+..  _rte-proc-entryHTMLparser_rte:
 
 entryHTMLparser_rte
 ~~~~~~~~~~~~~~~~~~~
 
-:aspect:`Datatype`
-    boolean / :ref:`HTMLparser <t3tsref:htmlparser>`
+..  confval:: proc.entryHTMLparser_rte
+    :name: rte-proc-entryHTMLparser_rte:
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:441]:
```suggestion
    :name: rte-proc-entryHTMLparser_rte
```

File: Documentation/PageTsconfig/Rte.rst [hunk 21]
```diff
@@ -414,151 +340,156 @@ allowedClasses
...
+    RTE; HTMLparser DB exit
+..  _rte-proc-exitHTMLparser_db:
 
 exitHTMLparser_db
 ~~~~~~~~~~~~~~~~~
 
-:aspect:`Datatype`
-    boolean / :ref:`HTMLparser <t3tsref:htmlparser>`
+..  confval:: proc.exitHTMLparser_db
+    :name: rte-proc-exitHTMLparser_db:
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:457]:
```suggestion
    :name: rte-proc-exitHTMLparser_db
```

File: Documentation/PageTsconfig/Rte.rst [hunk 22]
```diff
@@ -414,151 +340,156 @@ allowedClasses
...
+    RTE; HTMLparser RTE exit
+..  _rte-proc-exitHTMLparser_rte:
 
 exitHTMLparser_rte
 ~~~~~~~~~~~~~~~~~~
 
-:aspect:`Datatype`
-    boolean / :ref:`HTMLparser <t3tsref:htmlparser>`
+..  confval:: proc.exitHTMLparser_rte
+    :name: rte-proc-exitHTMLparser_rte:
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:473]:
```suggestion
    :name: rte-proc-exitHTMLparser_rte
```

File: Documentation/PageTsconfig/Rte.rst [hunk 23]
```diff
@@ -414,151 +340,156 @@ allowedClasses
...
+    RTE; HTMLparser DB
+..  _pageTsRteProcHtmlParserDb:
 
 HTMLparser_db
 ~~~~~~~~~~~~~
 
-:aspect:`Datatype`
-    :ref:`HTMLparser <t3tsref:htmlparser>`
+..  confval:: proc.HTMLparser_db
+    :name: rte-proc-HTMLparser_db:
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:490]:
```suggestion
    :name: rte-proc-HTMLparser_db
```

File: Documentation/PageTsconfig/Rte.rst [hunk 24]
```diff
@@ -589,41 +522,42 @@ files <t3coreapi:configuration-files>`
...
+    RTE; HTMLparser RTE
+..  _rte-proc-HTMLparser_rte:
 
 HTMLparser_rte
 ~~~~~~~~~~~~~~
 
-:aspect:`Datatype`
-    :ref:`HTMLparser <t3tsref:htmlparser>`
+..  confval:: proc.HTMLparser_rte
+    :name: rte-proc-HTMLparser_rte:
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:558]:
```suggestion
    :name: rte-proc-HTMLparser_rte
```

File: Documentation/PageTsconfig/Rte.rst [hunk 25]
```diff
@@ -632,22 +566,23 @@ HTMLparser_rte
...
+    RTE; Transformations overruled
+..  _rte-proc-:
 
 overruleMode
 ~~~~~~~~~~~~
 
-:aspect:`Datatype`
-    Comma list of RTE transformations
+..  confval:: proc.overruleMode
+    :name: rte-proc-overruleMode:
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:585]:
```suggestion
    :name: rte-proc-overruleMode
```

**Review by brotkrueml — APPROVED** [review]

**Pull request merged**